packages/itwinui-react/src/core/RadioTiles/RadioTile.tsx

```diff
@@ -28,29 +57,82 @@ type RadioTileProps = {
  * <RadioTile label='My tile' description='Some info' icon={<SvgSmileyHappy />} />
  */
 export const RadioTile = React.forwardRef((props, ref) => {
-  const { icon, label, description, className, style, ...rest } = props;
+  const {
+    icon,
+    label,
+    style,
+    description,
+    setFocus = false,
+    tileProps = {},
+    inputProps = {},
+    contentProps = {},
+    iconProps = {},
+    labelProps = {},
+    subLabelProps = {},
+  } = props;
 
   const inputElementRef = React.useRef<HTMLInputElement>(null);
   const refs = useMergedRefs<HTMLInputElement>(inputElementRef, ref);
 
+  React.useEffect(() => {
+    if (inputElementRef.current && setFocus) {
+      inputElementRef.current.focus();
+    }
+  }, [setFocus]);
+
+  const { className: tileClassName, ...restTileProps } = tileProps;
+  const { className: inputClassName, ...restInputProps } = inputProps;
+  const { className: contentClassName, ...restContentProps } = contentProps;
+  const { className: iconClassName, ...restIconProps } = iconProps;
+  const { className: labelClassName, ...restLabelProps } = labelProps;
+  const { className: subLabelClassName, ...restSubLabelProps } = subLabelProps;
```

Comment:
could you use the less verbose approach by _not_ destructuring? see [`fc3e8d6`](https://github.com/iTwin/iTwinUI/pull/1407/commits/fc3e8d6bb4888ad5a27fc633b24803d61a3857e0)

Reply:
done

Reply:
only the `className` prop needs to be moved to the end, the others can stay as before.

for example, in this case, `as='input'` and `type='radio'` should be moved to the lines before `...inputProps`. this will ensure that `inputProps` can override them.
<img width="575" alt="image" src="https://github.com/iTwin/iTwinUI/assets/9084735/f715bef5-a26b-4bb0-a7b9-fe6f188edc59">

same thing should be done for all elements.

Reply:
Done

Reply:
`disabled` should also be moved up